### DISCUSSION TITLE
Weird Ui displacement in 0.12

### DISCUSSION THREAD

**Discussion by Litttlefish** [bevyengine/bevy#10381] [Q&A]
Today I updated bevy to 0.12, and after migration I found this weird UI displacement
![图片](https://github.com/bevyengine/bevy/assets/38809254/ce781539-3bec-4d90-97a5-31098fd96ffa)
In 0.11, the ui works perfectly.
![图片](https://github.com/bevyengine/bevy/assets/38809254/b85aba30-04b4-4db6-a1e9-4470d747f520)

It happens when window resizes, also the ui that should be centered will also shift left by 1px.

**Comment by Litttlefish** [comment]
I made a minimal reproduce code

for sprite, any 16:9 image is ok
and maximize it after run to see the bug at the right side

```
use bevy::prelude::*;
use bevy::render::camera::ScalingMode;
use bevy::window::*;

fn main() {
    App::new()
        .add_systems(Startup, setup)
        .add_systems(Update, (setres.run_if(on_event::<WindowResized>()),))
        .add_plugins((DefaultPlugins.set(WindowPlugin {
            primary_window: Some(Window {
                title: "test".into(),
                resolution: (960.0, 540.0).into(),
                resizable: false,
                ..default()
            }),
            ..default()
        }),))
        //.add_plugin(EditorPlugin::default())
        .run();
}

fn setup(mut cmd: Commands, asset_server: Res<AssetServer>) {
    cmd.spawn(Camera2dBundle {
        projection: OrthographicProjection {
            scaling_mode: ScalingMode::AutoMin {
                min_width: 960.,
                min_height: 540.,
            },
            ..Camera2dBundle::default().projection
        },
        ..default()
    });
    cmd.spawn((SpriteBundle {
        sprite: Sprite {
            custom_size: Some((960., 540.).into()),
            ..default()
        },
        texture: asset_server.load("sprites/bedroom_normal.png"),
        ..default()
    },));
    cmd.spawn((NodeBundle {
        style: Style {
            width: Val::Percent(100.),
            height: Val::Percent(100.),
            justify_content: JustifyContent::Center,
            align_items: AlignItems::Center,
            ..default()
        },
        ..default()
    },))
        .with_children(|p| {
            p.spawn(NodeBundle {
                style: Style {
                    width: Val::Px(960.),
                    height: Val::Px(540.),
                    ..default()
                },
                background_color: BackgroundColor(Color::BLACK),
                ..default()
            });
        });
}

fn setres(mut scaler: ResMut<UiScale>, windowq: Query<&Window, With<PrimaryWindow>>) {
    scaler.0 = (windowq.single().height() as f64 / 540.).max(0.1)
}

**Reply by Litttlefish** [reply]
Should look like this
![图片](https://github.com/bevyengine/bevy/assets/38809254/b00e432b-7650-4259-a155-b8d437d92508)

As you can see, the right side has one pixel missing

**Comment by Litttlefish** [comment]
Also it seems that the displacement sometimes will change depending on resolution and presumably UiScale
On maximum(UiScale is about 1.9)
![image-4](https://github.com/bevyengine/bevy/assets/38809254/5dd014c0-df29-48e2-80b3-ab6b485f7dbf)
On fullscreen(UiScale is 2.0)
![image-5](https://github.com/bevyengine/bevy/assets/38809254/420332ca-7eea-426c-8651-08e4b3ee6140)

**Comment by ickshonpe** [comment]
It looks like another problem with coordinate rounding. 
Could you make us a minimal example that reproduces the error and post it here?

**Reply by Litttlefish** [reply]
Actually the minimal example here for me is:
create a background with 960*540px, window set to AutoMin, with UiScale
then create a node with 
`        NodeBundle {
            style: Style {
                width: Val::Percent(100.),
                height: Val::Percent(100.),
                justify_content: JustifyContent::Center,
                align_items: AlignItems::Center,
                ..default()
            },
            ..default()
        },`
and with children
`        NodeBundle {
            style: Style {
                width: Val::Px(960.),
                height: Val::Px(540.),
                ..default()
            },
            background_color: BackgroundColor(Color::BLACK),
            ..default()
        },`
If everything works out, the right side will have a line, which is not covered
I'll write a cleaner demo right up

**Comment by Litttlefish** [comment]
I think I should open it as an issue?🤔
